addons/website_sale_collect/tests/test_payment.py

```diff
@@ -26,3 +27,16 @@ def test_on_site_provider_unavailable_when_no_in_store_delivery(self):
         self.assertTrue(not any(
             p.code == 'custom' and p.custom_mode == 'on_site' for p in compatible_providers
         ))
+
+    def test_on_site_provider_tx_confirms_order(self):
+        order = self._create_so(carrier_id=self.carrier.id)
```

Comment:
IMO, it would be good to explicitly specify `state='draft'` to make it clear that the post-processing changes the state.